tools/PowerShell/Microsoft.WinGet.Client/src/Library/Install-WinGetPackage.ps1

```diff
@@ -0,0 +1,174 @@
+
+        .EXAMPLE
+        Install-WinGetPackage -id "Publisher.Package"
+
+        This example expects only a single package containing "Publisher.Package" as a valid identifier.
+
+        .EXAMPLE
+        Install-WinGetPackage -id "Publisher.Package" -source "Private"
+
+        This example expects the source named "Private" contains a package with "Publisher.Package" as a valid identifier.
+
+        .EXAMPLE
+        Install-WinGetPackage -Name "Package"
+
+        This example expects the source named "Private" contains a package with "Package" as a valid name.
+    #>
+
+    PARAM(
+        [Parameter(Position=0)] $Filter,
+        [Parameter()]           $Name,
+        [Parameter()]           $Id,
+        [Parameter()]           $Moniker,
+        [Parameter()]           $Source,
+        [Parameter()] [ValidateSet("User", "Machine")] $Scope,
+        [Parameter()] [switch]  $Interactive,
+        [Parameter()] [switch]  $Silent,
+        [Parameter()] [string]  $Version,
+        [Parameter()] [switch]  $Exact,
+        [Parameter()] [switch]  $Override,
+        [Parameter()] [System.IO.FileInfo]  $Location,
+        [Parameter()] [switch]  $Force,
+        [Parameter()] [ValidatePattern("^([a-zA-Z]{2,3}|[iI]-[a-zA-Z]+|[xX]-[a-zA-Z]{1,8})(-[a-zA-Z]{1,8})*$")] [string] $Locale,
+        [Parameter()] [string]  $Log, ## This is a path of where to create a log.
+        [Parameter()] [switch]  $AcceptSourceAgreements
+    )
+    BEGIN
+    {
+        [string[]] $WinGetArgs  = "Install"
+        IF($Filter){
+            $WinGetArgs += "--Manifest", $Filter
+        }
+        IF($PSBoundParameters.ContainsKey('Name')){
+            $WinGetArgs += "--Name", $Name
+        }
+        IF($Id){
+            $WinGetArgs += "--Id", $Id
+        }
+        IF($Moniker){
+            $WinGetArgs += "--Moniker", $Moniker
+        }
+        IF($Source){
+            $WinGetArgs += "--Source", $Source
+        }
+        IF($Scope){
+            $WinGetArgs += "--Scope", $Scope
+        }
+        IF($Interactive){
+            $WinGetArgs += "--Interactive"
+        }
+        IF($Silent){
+            $WinGetArgs += "--Silent"
+        }
+        IF($Locale){
+            $WinGetArgs += "--locale", $Locale
+        }
+        if($Version){
+            $WinGetArgs += "--Version", $Version
+        }
+        if($Exact){
+            $WinGetArgs += "--Exact"
+        }
+        if($Log){
+            $WinGetArgs += "--Log", $Log
+        }
+        if($Override){
+            $WinGetArgs += "--override", $Override
+        }
+        if($Location){
+            $WinGetArgs += "--Location", $Location
+        }
+        if($Force){
+            $WinGetArgs += "--Force"
+        }
+    }
+    PROCESS
+    {
+        ## Exact, ID and Source - Talk with Demitrius tomorrow to better understand this.
+        $Result = Find-WinGetPackage -Filter $Filter -Name $Name -Id $Id -Moniker $Moniker -Tag $Tag -Command $Command -Source $Source
```

Comment:
Not sure that this will work if any of these values are null. Unable to test - see other comments on Find-WingetPackage